frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/LogsTab.tsx

```diff
@@ -0,0 +1,186 @@
+import useFetchLogs from '~/concepts/k8s/pods/useFetchLogs';
+import usePodContainerState from '~/concepts/k8s/pods/usePodContainerState';
+import LogsTabStatus from '~/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/LogsTabStatus';
+import { LOG_TAIL_LINES } from '~/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/const';
+import { downloadString } from '~/utilities/string';
+import DashboardCodeEditor from '~/concepts/dashboard/codeEditor/DashboardCodeEditor';
+import useCodeEditorAsLogs from '~/concepts/dashboard/codeEditor/useCodeEditorAsLogs';
+import { downloadFullPodLog } from '~/concepts/k8s/pods/utils';
+import { usePipelinesAPI } from '~/concepts/pipelines/context';
+
+type LogsTabProps = {
+  task: PipelineRunTaskDetails;
+};
+
+const NO_POD_ERROR = new Error('No pod found, unable to show logs');
+
+const LogsTab: React.FC<LogsTabProps> = ({ task }) => {
+  const { namespace } = usePipelinesAPI();
+  const podName = task.runDetails?.status.podName ?? '';
+  const podNameError = !podName ? NO_POD_ERROR : undefined;
+  const { podLoaded, podError, podContainers, selectedContainer, setSelectedContainer } =
+    usePodContainerState(podName);
+  const [isPaused, setIsPaused] = React.useState(false);
+  const [logs, logsLoaded, logsError, refreshLogs] = useFetchLogs(
+    podName,
+    selectedContainer?.name ?? '',
+    !isPaused,
+    LOG_TAIL_LINES,
+  );
+  const [downloading, setDownloading] = React.useState(false);
+  const [downloadError, setDownloadError] = React.useState<Error | undefined>();
+  const { scrollToBottom, onMount, editorOptions } = useCodeEditorAsLogs();
+
+  const scrollEditorToBottom = React.useCallback<typeof scrollToBottom>(() => {
+    if (!isPaused) {
+      scrollToBottom();
+    }
+  }, [isPaused, scrollToBottom]);
+
+  React.useEffect(() => {
+    if (!isPaused && logs) {
+      scrollEditorToBottom();
+    }
+  }, [isPaused, logs, scrollEditorToBottom]);
+
+  const canDownload = !!selectedContainer && !!podName && !downloading;
+  const onDownload = () => {
+    if (!canDownload) {
+      return;
+    }
+    setDownloadError(undefined);
+    downloadFullPodLog(namespace, podName, selectedContainer.name)
+      .then(() => setDownloading(false))
+      .catch((e) => {
+        setDownloading(false);
+        setDownloadError(e);
+      });
+  };
+
+  const error = podError || podNameError || logsError || downloadError;
+  const loaded = podLoaded && logsLoaded;
+
+  let data: string;
+  if (!error && !logsLoaded) {
+    data = 'Loading...';
+  } else if (logs) {
+    data = logs;
+  } else {
+    data = 'No logs available';
+  }
+
+  return (
+    <Stack hasGutter>
+      <StackItem>
+        <LogsTabStatus
+          loaded={loaded}
+          error={error}
+          refresh={refreshLogs}
+          onDownload={onDownload}
+          canDownload={canDownload}
+        />
+      </StackItem>
+      <StackItem>
+        <Split hasGutter isWrappable>
```

Comment:
This should be another component -- I'll see if I have enough time to do so tomorrow before we merge it.